src/pepr/operator/controllers/keycloak/client-sync.ts

```diff
@@ -65,24 +64,25 @@ export async function keycloak(pkg: UDSPackage) {
  * @param pkg the package to process
  * @param refs the list of client refs to keep
  */
-export async function purgeSSOClients(pkg: UDSPackage, refs: string[] = []) {
+export async function purgeSSOClients(pkg: UDSPackage, newClients: string[] = []) {
   // Check for any clients that are no longer in the package and remove them
   const currentClients = pkg.status?.ssoClients || [];
-  const toRemove = currentClients.filter(client => !refs.includes(client));
+  const toRemove = currentClients.filter(client => !newClients.includes(client));
   for (const ref of toRemove) {
-    const token = Store.getItem(ref);
-    const clientId = ref.replace("sso-client-", "");
+    const storeKey = `sso-client-${ref}`;
+    const token = Store.getItem(storeKey);
     if (token) {
-      Store.removeItem(ref);
-      await apiCall({ clientId }, "DELETE", token);
+      Store.removeItem(storeKey);
+      await apiCall({ clientId: ref }, "DELETE", token);
```

Comment:
Not a change in this PR but maybe worth considering swapping the order of these two so that we don't lose a token if the deletion fails? Unsure if we can really recover in that case, but probably best to reorder?

Reply:
Good call.  Will update